scripts/reloader.bash

```diff
@@ -1,54 +1,81 @@
-#!/bin/bash
+#!/usr/bin/env bash
+# shellcheck disable=SC1090,SC2221,SC2222
+
 BASH_IT_LOG_PREFIX="core: reloader: "
-pushd "${BASH_IT}" >/dev/null || exit 1
+pushd "${BASH_IT}" > /dev/null || exit 1
+
+_bash-it-load-sources() {
+	if [[ -z "$1" ]]; then
+		return 0
+	fi
+
+	local scripts current output retval
+
+	scripts=("$@")
+	current="${scripts[0]}"
+	scripts=("${scripts[@]:1}")
+
+	filename=$(_bash-it-get-component-name-from-path "$current")
+	extension=$(_bash-it-get-component-type-from-path "$current")
+	BASH_IT_LOG_PREFIX="$extension: $filename: "
+
+	if [[ -e $current ]]; then
 
-function _set-prefix-based-on-path()
-{
-  filename=$(_bash-it-get-component-name-from-path "$1")
-  extension=$(_bash-it-get-component-type-from-path "$1")
-  BASH_IT_LOG_PREFIX="$extension: $filename: "
+		_log_debug "Loading component..."
+		output=$(
+			set -e
+			(source "$current") 2>&1
+		)
+		retval=$?
+		set +e
+
+		if [[ $retval -gt 0 ]]; then
+			_log_error 'Loading component failed...'
+			if [[ -n "$output" ]]; then
+				while read -r line; do _log_error "$line"; done <<< "$output"
+			fi
+		else
+			_log_debug 'Loading component successful!'
+			source "$current"
+			if [[ -n "$output" ]]; then
+				while read -r line; do _log_debug "$line"; done <<< "$output"
```

Comment:
this also causes log messages in the plugin to be displayed twice (once regularly and once by this line)
we currently print logs to stdout, so maybe only catch output of stderr to display